### PR TITLE
DELETE and GET mappings for CommunityAmenity

### LINKED ISSUE #110: Endpoint for getting details of an amenity
## ⚠️ Is your feature request related to a problem? Please describe

* Get details of an amenity

## 💡 Describe the solution you'd like

* `GET /amenities/{amenityId}`

## Task Relationship

Subtask of: jmprathab/MyHome#93

### LINKED ISSUE #112: Endpoing for deleting an amenity
## ⚠️ Is your feature request related to a problem? Please describe

* REST endpoint which deletes an amenity.

## 💡 Describe the solution you'd like

* `DELETE /amenities/{amenityId}`

## Task Relationship

Subtask of: jmprathab/MyHome#93

### PR DESCRIPTION
**Pull request by DDmit04**
## 🚀 Description

Add GET and DELETE (`/amenities/{amenityId}`) endpoints for `CommunityAmenity` 
Also I update postman config with new enpoints

## 📄 Motivation and Context

Closes #112 
Closes #110 

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by DDmit04** [comment]
this commit d4199e1 solve part of issue #123

File: src/main/java/com/myhome/controllers/CommunityAmenityController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  @Operation(
+      description = "Get details about the amenity",
+      responses = {
+          @ApiResponse(responseCode = "404", description = "If params are invalid"),
+      }
+  )
+  @GetMapping("/amenities/{amenityId}")
+  public ResponseEntity getAmenityDetails(@PathVariable String amenityId) {
+    Optional<CommunityAmenity> communityAmenityOptional = communityAmenitySDJpaService.getCommunityAmenityDetails(amenityId);
+    return communityAmenityOptional
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/CommunityAmenityController.java]: We can change it to:
```
return communityAmenitySDJpaService.getCommunityAmenityDetails(amenityId)
             .map(communityAmenity -> {...})...
```

File: src/main/java/com/myhome/controllers/CommunityAmenityController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        .map(communityAmenity -> {
+          GetCommunityAmenityDetailsResponse response = communityAmenityApiMapper
+              .communityAmenityToCommunityAmenityDetailsResponse(communityAmenity);
+          return ResponseEntity.status(HttpStatus.OK).body(response);
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/CommunityAmenityController.java]: `ResponseEntity.status(HttpStatus.OK).body(response)` can be in a separate `map` operator

File: src/main/java/com/myhome/controllers/CommunityAmenityController.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+  @GetMapping("/communities/{communityId}/amenities}")
+  public ResponseEntity listAllCommunityAmenities(@PathVariable String amenityId) {
+    Optional<CommunityAmenity> communityAmenityOptional = communityAmenitySDJpaService.getCommunityAmenityDetails(amenityId);
+    return communityAmenityOptional
+        .map(communityAmenity -> {
+          GetCommunityAmenityDetailsResponse response = communityAmenityApiMapper
+              .communityAmenityToCommunityAmenityDetailsResponse(communityAmenity);
+          return ResponseEntity.status(HttpStatus.OK).body(response);
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/CommunityAmenityController.java]: `ResponseEntity.status(HttpStatus.OK).body(response)` can be in a separate map operator


`ResponseEntity.ok(response)` is shortcut for your approach

File: src/main/java/com/myhome/controllers/CommunityAmenityController.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      description = "Remove amenity",
+      responses = {
+          @ApiResponse(responseCode = "204", description = "If amenity deleted"),
+          @ApiResponse(responseCode = "404", description = "If params are invalid"),
+      }
+  )
+  @DeleteMapping(path = "/amenities/{amenityId}")
+  public ResponseEntity deleteAmenity(@PathVariable String amenityId) {
+    boolean isAmenityDeleted = communityAmenitySDJpaService.deleteAmenity(amenityId);
+    if(isAmenityDeleted) {
```
Comment (mslowiak) [inline src/main/java/com/myhome/controllers/CommunityAmenityController.java]: formatting

**Review by mslowiak — CHANGES_REQUESTED** [review]

**Comment by mslowiak** [comment]
We have updated the contributing guide with points 5 and 6.
Please take a look and apply the required changes:
[contributing.md](https://github.com/jmprathab/MyHome/blob/master/CONTRIBUTING.md)

**Comment by mslowiak** [comment]
And would be great if you can make one pull request per issue in the feature. Thanks :)

**Pull request merged**